### ISSUE TITLE
tf.grad() no longer works on Sequential models after 2.4.0

### ISSUE THREAD

**Issue by kennysong** [tensorflow/tfjs#4130]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 10.15.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow.js installed from (npm or script link): https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@2.4.0/dist/tf.min.js
- TensorFlow.js version (use command below): 2.4.0
- Browser version: Chrome 86.0.4240.80
- Tensorflow.js Converter Version: None

**Describe the current behavior**

`tf.grad()` works correctly on a Sequential model in 2.4.0.

**Describe the expected behavior**

`tf.grad()` fails on the same Sequential model in 2.5.0 and 2.6.0.

(In 2.5.0, I used `tf.env().set('WEBGL_CPU_FORWARD', false);` to [get the code running](https://github.com/tensorflow/tfjs/issues/4029#issuecomment-704832446).)

**Standalone code to reproduce the issue**

This simple example works on 2.4.0 but not on 2.5.0 and 2.6.0.

```
let model = tf.sequential();
model.add(tf.layers.dense({inputShape: [1], units: 1, activation: 'linear'}));

let f = x => model.predict(x);
let g = tf.grad(f);

let t = tf.tensor([1]);
f(t).print();
g(t).print();
```

It fails with this stack trace:

```
Uncaught TypeError: Cannot read property 'shape' of undefined
    at gradFunc (Concat_grad.js:29)
    at Object.s.gradient (engine.js:931)
    at a (tape.js:158)
    at tape.js:136
    at engine.js:1038
    at engine.js:433
    at e.t.scopedRun (engine.js:444)
    at e.t.tidy (engine.js:431)
    at e.t.gradients (engine.js:1033)
    at gradients.js:69
```

**Comment by lina128** [comment]
Fixed by https://github.com/tensorflow/tfjs/pull/4064

**Comment by tafsiri** [comment]
@kennysong note that this is fixed in 2.7.0.
